### PR TITLE
Deploy snapshots to GitHub packages.

### PR DESCRIPTION
**Pull request by mshima**
Related to https://github.com/jhipster/generator-jhipster/issues/12251

### CODE REVIEW

**Comment by DanielFran** [comment]
@jdubois We already discuss to deploy snapshots. Now that we can on GitHub it is the opportunity to do it...

**Comment by pascalgrimaud** [comment]
As mentioned in PR and at https://github.com/jhipster/generator-jhipster/issues/12251#issuecomment-674552308

I'm not totally convinced about this approach. But let's discuss

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        with:
+          java-version: 11
+      - name: Cache Maven archetypes
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+      - name: Deploy to github packages
```
Comment (pascalgrimaud) [inline .github/workflows/deploy.yml:44]: I propose some change:
- rename `sonar.yml` to something else, like `build.yml`
- add this new part in `build.yml` so we'll deploy only after mvn test and mvn sonar

**Comment by pascalgrimaud** [comment]
After thinking about it, we can give a try and deploy to GitHub package.
It won't have any impact directly, excepting if we decide to use `release` value + snapshot version.

**Comment by pascalgrimaud** [comment]
OK then, so let's give a try !
I'm merging this

**Pull request merged**